### PR TITLE
Added tablet support for onDragOverSquare function

### PR DESCRIPTION
**Pull request by KeeghanM**
On tablet, the onDragOverSquare function wasn't being called.
I added in an onTouchMove event on the square Div which now calls the onDragOverSquare if the square is different,

This maintains current behaviour of it only being called onEnter.

### CODE REVIEW

File: src/chessboard/components/Square.tsx [hunk 1]
```diff
@@ -113,6 +115,21 @@ export function Square({
       style={defaultSquareStyle}
       data-square-color={squareColor}
       data-square={square}
+      onTouchMove={(e) => {
+        // Handle touch events on tablet and mobile not covered by onMouseOver
```
[the remark this answers is not in the document]
Comment (KeeghanM) [reply src/chessboard/components/Square.tsx:119]: Both to be honest, neither of them are firing on tablets correctly

File: src/chessboard/components/Square.tsx [hunk 2]
```diff
@@ -69,6 +69,8 @@ export function Square({
       lastPieceColour,
     ]
   );
+  const [lastSquareDraggedOver, setLastSquareDraggedOver] =
+    useState<Sq | null>(null);
```
Comment (Manukyanq) [inline src/chessboard/components/Square.tsx:73]: Hmmm `Square` component represents one square. currently you have been added 64 independent states for each Square. Can you explain please why?

Comment (Clariity) [reply src/chessboard/components/Square.tsx:73]: this may be better placed in the chessboard context

Comment (KeeghanM) [reply src/chessboard/components/Square.tsx:73]: The simple solution is not do the check, but then you have the onDragOverSquare() running a million times, and I know in my own project that would kill it - a fair bit happens in that function.
I'll take a look at getting it moved into the Chessboard Context - I think you're right, probably works better there.

File: src/chessboard/components/Square.tsx [hunk 3]
```diff
@@ -51,6 +51,8 @@ export function Square({
     setPromoteFromSquare,
     setPromoteToSquare,
     setShowPromoteDialog,
+    lastSquareDraggedOver,
+    setLastSquareDraggedOver,
```
Comment (Manukyanq) [inline src/chessboard/components/Square.tsx:55]: all imports in project are in alphabetical order, not critical for me, but will be nice to keep that order

File: src/chessboard/context/chessboard-context.tsx [hunk 4]
```diff
@@ -102,6 +102,8 @@ interface ChessboardProviderContext {
   setPromoteToSquare: React.Dispatch<React.SetStateAction<Square | null>>;
   setShowPromoteDialog: React.Dispatch<React.SetStateAction<boolean>>;
   showPromoteDialog: boolean;
+  lastSquareDraggedOver: Square | null;
+  setLastSquareDraggedOver: React.Dispatch<React.SetStateAction<Square | null>>;
```
Comment (Manukyanq) [inline src/chessboard/context/chessboard-context.tsx:106]: same here

**Review by Clariity — APPROVED** [review]

**Pull request merged**